xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
If it works out better I can just ifndef some functions for X11 vs Mir as well. Ill try to take some time to looking at doing an implementation for Mir in vaapi.

Seems like currently its hardcoded for X11 + EGL, I dont have a multiple GPU on this machine but reading about render nodes they are independent of cards ie. any *should* work but I cant test that.

I would prefer to at lease has this DRM as backup support since I would assume most cases have a single GPU, which is better then software rendering.